greengrass_defender_agent/agent.py

```diff
@@ -67,33 +67,43 @@ def wait_for_config_changes(ipc_client):
     wait_for_config_changes(ipc_client)
 
 
-def publish_metrics(ipc_client, new_config, config_changed):
+def publish_metrics(ipc_client, new_config):
     """
-    Collect and publish metrics.
+    Initiate the metrics collector.
 
     :param ipc_client: Ipc client
     :param new_config: a configuration object
+    """
+    config.logger.info("Collector running on device: {}".format(config.THING_NAME))
+    config.logger.info("Metrics topic: {}".format(config.TOPIC))
+    sample_interval_seconds = new_config[config.SAMPLE_INTERVAL_NEW_CONFIG_KEY]
+    config.logger.info("Sampling interval: {} seconds".format(sample_interval_seconds))
+
+    metrics_collector = collector.Collector(short_metrics_names=False)
```

Comment:
Even if config changed, none of the config affects this object so don't create it again. Create it only 1 time in the constrictor of a class

Reply:
Updated. 
I had weighed both options. Was thinking the user might want everything "refreshed" when changing the configuration. And the first delta after a reconfiguration would not represent the delta for a full interval. But after reconsidering I agree it makes sense to only have one collector object. The system metrics' delta should preserve despite configuration change.